### ISSUE TITLE
Failed to install: wrond size for extra data

### ISSUE THREAD

**Issue by robbyneven** [flathub/com.jagex.RuneScape#172]
Error: Wrong size for extra data https://content.runescape.com/downloads/ubuntu/pool/non-free/r/runescape-launcher/runescape-launcher_2.2.9_amd64.deb
error: Failed to install com.jagex.RuneScape: Wrong size for extra data https://content.runescape.com/downloads/ubuntu/pool/non-free/r/runescape-launcher/runescape-launcher_2.2.9_amd64.deb

**Comment by weedmic** [comment]
I believe this is caused each time a new client is introduced.  Ad hoc, 2.2.9.

Previously, I created a copy of .../stable directory with hidden files and created the following script:

#!/bin/sh
rm -rf /var/lib/flatpak/app/com.jagex.RuneScape/x86_64/stable
cp -r /home/weedmic/Downloads/runescape-savepoint/stable /var/lib/flatpak/app/com.jagex.RuneScape/x86_64/

which would allow me to use runescape the way it was (2.2.4) at least once until flatpak caught up.

I am unsure exactly what "causes" this, but I believe that there is a part of the script pointing to a wrong file - or the SHA is not updated to the new file.

In this particular instance 2.2.9, nothing works for me and I also get the following when doing "flatpak install flathub com.jagex.RuneScape"

"error: Failed to install com.jagex.RuneScape/x86_64/stable: Wrong size for extra data https://content.runescape.com/downloads/ubuntu/pool/non-free/r/runescape-launcher/runescape-launcher_2.2.9_amd64.deb"

This sounds similar to what I read on the r/s forum @ QFC 278-279-573-66145484

Seems someone at flathub and jagex just need to communicate.

I hope this will bring it to someone's attention.

**Comment by twofab** [comment]
tl;dr: problem is still there for 2.2.10
although the size and sha256 sum of the deb package do match those in https://github.com/flathub/com.jagex.RuneScape/blob/master/com.jagex.RuneScape.json

Today's runescape update broke the flatpak launcher for me (it would crash a few seconds into the login screen). I tried updating the flatpak and it raised the 'wrong size' error mentioned by robyneven. At that point there was a pull request by [flathubbot](https://github.com/flathubbot) to change the size and sha sum in the json, so I thought that it would fix it once the PR would go through. In the meantime, to be able to run the game I installed the test branch created by the bot (it's not available anymore unfortunately). There was no problem installing this test branch, and the game runs perfectly with it.
Then the PR was merged into master, and I tried updating again and it did not work, with the same error. I tried uninstalling, purging the data, and reinstalling and it still does not work...
So now I do not have a launcher from the stable branch anymore, only the test branch which cannot be updated.
